ttnn/cpp/ttnn/operations/data_movement/expand/device/expand_rm_program_factory.cpp

```diff
@@ -0,0 +1,213 @@
+
+    std::stringstream debug_stream;
+
+    debug_stream << "Input Shape = ";
+    for (auto i = 0; i < input_shape.size(); i++) {
+        debug_stream << input_shape[i] << " ";
+    }
+    debug_stream << std::endl;
+
+    debug_stream << "Output Shape = ";
+    for (auto i = 0; i < output_shape.size(); i++) {
+        debug_stream << output_shape[i] << " ";
+    }
+    debug_stream << std::endl;
+
+    tt::log_debug("%s", debug_stream.str().c_str());
+
+    tt::log_debug("Horz Expand Ratio = %d\n", horz_expand_count);
+    tt::log_debug("Vert Expand Ratio = %d\n", nd_expand_count);
+#endif
+
+    auto compute_with_storage_grid_size = device->compute_with_storage_grid_size();
+    uint32_t num_cores_x = compute_with_storage_grid_size.x;
+    uint32_t num_cores_y = compute_with_storage_grid_size.y;
+    auto [num_cores, all_cores, core_group_1, core_group_2, num_copies_per_core_group_1, num_copies_per_core_group_2] =
+        split_work_to_cores(compute_with_storage_grid_size, nd_expand_count);
+
+#ifdef DEBUG
+    tt::log_debug("Num Cores = %d\n", num_cores);
+    tt::log_debug("Num Rows Per Core Group 1 = %d\n", num_copies_per_core_group_1);
+    tt::log_debug("Num Rows Per Core Group 2 = %d\n", num_copies_per_core_group_2);
+#endif
+
+    const auto src_is_dram = static_cast<const uint32_t>(input.buffer()->is_dram());
+    const auto dst_is_dram = static_cast<const uint32_t>(output.buffer()->is_dram());
+
+    const auto sram_buffer_length = 32;
+
+    // Scratch SRAM buffer
+    uint32_t scratch_buf_id = tt::CB::c_intermed0;
+    auto scratch_config =
+        CircularBufferConfig(unexpanded_row_size * sram_buffer_length, {{scratch_buf_id, data_format}})
+            .set_page_size(scratch_buf_id, unexpanded_row_size);
+    auto scratch_handle = CreateCircularBuffer(program, all_cores, scratch_config);
+
+    // IO SRAM Buffer
+    uint32_t io_buf_id = tt::CB::c_out0;
+    auto io_config = CircularBufferConfig(expanded_row_size * sram_buffer_length, {{io_buf_id, data_format}})
+                         .set_page_size(io_buf_id, expanded_row_size);
+    auto io_handle = CreateCircularBuffer(program, all_cores, io_config);
+
+    std::vector<uint32_t> reader_compile_runtime_args = {
+        src_is_dram,
+        scratch_buf_id,
+        io_buf_id,
+        data_size,
+    };
+    std::vector<uint32_t> writer_compile_runtime_args = {
+        dst_is_dram,
+        io_buf_id,
+    };
+
+    KernelHandle reader_id = CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/data_movement/expand/device/kernels/reader_rm_expand.cpp",
+        all_cores,
+        ReaderDataMovementConfig(reader_compile_runtime_args));
+
+    KernelHandle writer_id = CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/data_movement/expand/device/kernels/writer_rm_expand.cpp",
+        all_cores,
+        WriterDataMovementConfig(writer_compile_runtime_args));
+
+    auto rows_offset = 0;
```

Comment:
```suggestion
    uint32_t rows_offset = 0;
```
This is probably an `int`, while the runtime argument is likely `uint32_t`.

Reply:
I've changed it to explicitly use uint32_t